tests/jaxite/end_to_end/add_one_lut3.mlir

```diff
@@ -26,38 +15,38 @@ func.func @test_add_one_lut3(%arg0: tensor<8x!eb>, %bsks : !bsks, %params : !par
-  %tt_120 = arith.constant 120 : i8
-  %tt_128 = arith.constant 128 : i8
-
-  %constant_T = jaxite.constant %bool_1, %params : (i1, !params) -> !eb
-  %constant_F = jaxite.constant %bool_0, %params : (i1, !params) -> !eb
-
-  %t_0 = jaxite.lut3 %x_00, %x_01, %x_02, %tt_128, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %t_1 = jaxite.lut3 %t_0, %x_03, %x_04, %tt_128, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %t_2 = jaxite.lut3 %t_1, %x_05, %x_06, %tt_128, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-
-  %res_07 = jaxite.lut3 %t_2, %x_07, %constant_F, %tt_6, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %res_06 = jaxite.lut3 %t_1, %x_05, %x_06, %tt_120, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %res_05 = jaxite.lut3 %t_1, %x_05, %constant_F, %tt_6, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %res_04 = jaxite.lut3 %t_0, %x_03, %x_04, %tt_120, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %res_03 = jaxite.lut3 %t_0, %x_03, %constant_F, %tt_6, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %res_02 = jaxite.lut3 %x_00, %x_01, %x_02, %tt_120, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %res_01 = jaxite.lut3 %x_00, %x_01, %constant_F, %tt_6, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-  %res_00 = jaxite.lut3 %x_00, %constant_F, %constant_F, %tt_1, %bsks, %params : (!eb, !eb, !eb, i8, !bsks, !params) -> !eb
-
-  %from_elements = tensor.from_elements %res_00, %res_01, %res_02, %res_03, %res_04, %res_05, %res_06, %res_07 : tensor<8x!eb>
-  return %from_elements : tensor<8x!eb>
+
+
+  %encoded_1 = lwe.encode %bool_1 {encoding = #unspecified_encoding} : i1 to !pt_ty
+  %constant_T = lwe.trivial_encrypt %encoded_1 : !pt_ty to !ct_ty
+  %encoded_0 = lwe.encode %bool_0 {encoding = #unspecified_encoding} : i1 to !pt_ty
+  %constant_F = lwe.trivial_encrypt %encoded_0 : !pt_ty to !ct_ty
+
+  %t_0 = cggi.lut3 %x_00, %x_01, %x_02 {lookup_table = 128 : ui8} : !ct_ty
+  %t_1 = cggi.lut3 %t_0, %x_03, %x_04 {lookup_table = 128 : ui8} : !ct_ty
+  %t_2 = cggi.lut3 %t_1, %x_05, %x_06 {lookup_table = 128 : ui8} : !ct_ty
+
+
+  %res_07 = cggi.lut3 %t_2, %x_07, %constant_F {lookup_table = 6 : ui8} : !ct_ty
+  %res_06 = cggi.lut3 %t_1, %x_05, %x_06 {lookup_table = 120 : ui8} : !ct_ty
+  %res_05 = cggi.lut3 %t_1, %x_05, %constant_F {lookup_table = 6 : ui8} : !ct_ty
+  %res_04 = cggi.lut3 %t_0, %x_03, %x_04 {lookup_table = 120 : ui8} : !ct_ty
+  %res_03 = cggi.lut3 %t_0, %x_03, %constant_F {lookup_table = 6 : ui8} : !ct_ty
+  %res_02 = cggi.lut3 %x_00, %x_01, %x_02 {lookup_table = 120 : ui8} : !ct_ty
+  %res_01 = cggi.lut3 %x_00, %x_01, %constant_F {lookup_table = 6 : ui8} : !ct_ty
+  %res_00 = cggi.lut3 %x_00, %constant_F, %constant_F {lookup_table =1 : ui8} : !ct_ty
+
+  %from_elements = tensor.from_elements %res_00, %res_01, %res_02, %res_03, %res_04, %res_05, %res_06, %res_07 : tensor<8x!ct_ty>
```

Comment:
nice! when you actually run add_one through the yosys pipeline it'll use memref semantics instead of tensor, so expect to add memref ops to this conversion pass when you do that.

Reply:
Sg. Memref PR coming up after this.